torch/utils/checkpoint.py

```diff
@@ -70,7 +70,7 @@ def forward(ctx, run_function, preserve_rng_state, *args):
             if torch.cuda._initialized:
                 ctx.had_cuda_in_fwd = True
                 ctx.fwd_gpu_devices, ctx.fwd_gpu_states = get_device_states(*args)
-        ctx.save_for_backward(*args)
+        ctx.inputs = args
```

Comment:
Would the tensor inputs still be freed on the backward pass here when `retain_graph=False`?
Maybe we could pick out the tensor arguments from inputs and still use `save_for_backward` in those cases and only use ctx to store non-tensor inputs